gp-includes/misc.php

```diff
@@ -331,3 +331,40 @@ function gp_gmt_strtotime( $string ) {
 
 	return $time;
 }
+
+/**
+ * Determines the format to use based on the selected format type or by auto detection based on the file name.
+ *
+ * Used during import of translations and originals.
+ *
+ * @param string $selected_format The format that the user selected on the import page.
+ * @param string $filename The filname that was uploaded by the user.
+ * @return object A GP_Format child object.
+ */
+function gp_get_import_file_format( $selected_format, $filename ) {
+	$format = gp_array_get( GP::$formats, $selected_format, null );
+
+	if ( ! $format ) {
+		$extension = pathinfo( $filename, PATHINFO_EXTENSION );
+		
+		foreach( GP::$formats as $format_entry ) {
+			if( $extension == $format_entry->extension ) {
```

Comment:
This could be simplified to:
```php
$format_extensions = array_merge( array( $format_entry->extension ), $format_entry->alt_extensions );
if ( in_array( $extension, $format_extensions ) {
    $format = $format_entry;
}
```